### PR TITLE
Added detector volumes

### PR DESCRIPTION
**Pull request by WenzDaniel**
Since I fucked up with the other branch, we decided to make a new branch from master and do the changes one by one.

### CODE REVIEW

File: bin/run_epix [hunk 1]
```diff
@@ -1,122 +1,133 @@
...
-    print(f'Reading in root file {file_name}')
-    inter = epix.loader(path, file_name, kwargs_uproot_ararys={'entry_stop': args.EntryStop})
-    if args.Timing:
-        t = time.time()
-        print(f'It took {round(t - tnow, 5)} sec to load the data.')
-        tnow = t
-
-    print((f'Finding clusters of interactions with a dr = {args.MicroSeparation} mm'
-           f' and dt = {args.MicroSeparationTime} ns'))
-    inter = epix.find_cluster(inter, args.MicroSeparation/10, args.MicroSeparationTime)
```
Comment (WenzDaniel) [inline bin/run_epix:82]: Should not we just change the corresponding option to cm?

File: bin/run_epix [hunk 2]
```diff
@@ -1,122 +1,133 @@
...
+                        kwargs_uproot_ararys={'entry_stop': args.EntryStop}
+                        )
+
+    if args.debug:
+        tnow = monitor_time(tnow, 'load data.')
+        print((f'Finding clusters of interactions with a dr = {args.MicroSeparation} cm'
+               f' and dt = {args.MicroSeparationTime} ns'))
+
+    # Cluster finding and clustering:
+    inter = epix.find_cluster(inter, args.MicroSeparation, args.MicroSeparationTime)
```
Comment (WenzDaniel) [inline bin/run_epix:79]: See comment above

File: bin/run_epix (hunk 1 above)
Comment (ramirezdiego) [reply bin/run_epix:82]: I find that, for this particular input, units in mm or microns are more intuitive for the user.

File: epix/detector_volumes.py [hunk 3]
```diff
@@ -1,129 +1,222 @@
...
+            if ak.any(m):
+                overlapping_id = result[m][0]
+                # Get volume name:
+                name = [vol.name for vol in sensitive_volumes if vol.volume_id == overlapping_id][0]
+                raise ValueError(f'The volume {vol.name} is overlapping with'
+                                 f' volume {name}!')
+            new_results = res.snapshot()
+            for field in result.fields():
+                # Workaround since we cannot sum up records-arrays anymore
+                result[field] = result[field] + new_results[field]
```
Comment (ramirezdiego) [inline epix/detector_volumes.py:188]: Shouldn't all the `result` in these lines be `res`? Otherwise I am missing what you do here.

File: epix/detectors.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                       'roi': _make_roi_cylinder(xenonnt_z_cathode,
+                                                 xenonnt_z_gate_mesh,
+                                                 xenonnt_sensitive_volume_radius),
+                       'electric_field': 200,
+                       'create_S2': True,
+                       'xe_density': 2.862,
+                       'efield_outside_map': 200,
+                       'to_be_stored': True,
+                       },
+               'BelowCathode': {'vol_id': 1,
```
Comment (HenningSE) [inline epix/detectors.py:44]: Here you have the same `vol_id` for the TPC and BelowCathode. Should this be changed to `2`?

Comment (ramirezdiego) [reply epix/detectors.py:44]: Sí!

File: bin/run_epix [hunk 5]
```diff
@@ -1,122 +1,133 @@
...
     # Sort detector volumes and keep interactions in selected ones:
-    sensitive_volumes = [epix.tpc, epix.below_cathode] #TODO: add options
-    print('Removing clusters not in volumes:', *[x.name for x in sensitive_volumes])
-    print(f'Number of clusters before: {np.sum(ak.num(result["x"]))}')
-    result['ids'] = epix.in_sensitive_volume(result, sensitive_volumes)
-    m = (result['ids'] == sensitive_volumes[0].volume_id) | (result['ids'] == sensitive_volumes[1].volume_id)
-    # TODO: The idea is to code this properly and depending on len(sensitive_volumes)
+    if args.debug:
+        print('Removing clusters not in volumes:', *[v.name for v in detector])
+        print(f'Number of clusters before: {np.sum(ak.num(result["ed"]))}')
```
Comment (ramirezdiego) [inline bin/run_epix:95]: The `Removing clusters not in volumes` output isn't entirely true, since we then list all the volumes defined for a detector in the detectors file, and not these in which we set the `to_be_stored` option to true.

Comment (WenzDaniel) [reply bin/run_epix:95]: `detectors` should only contain the volumes which have `to_be_stored=True`

Comment (ramirezdiego) [reply bin/run_epix:95]: I get the list of all implemented volumes, when loading the `example_xenonnt.ini` file (which has two volumes set to `False`).

Comment (ramirezdiego) [reply bin/run_epix:95]: Okay, I was setting the old config file name. This means epix doesn't complain if you give it a config filename that doesn't exist under `--Config`.

Comment (WenzDaniel) [reply bin/run_epix:96]: Okay solved, also added test for empty data/no interactions left.

**Comment by ramirezdiego** [comment]
When using `--Config config/example_xenonnt.ini`, I guet
```
io.py:38: UserWarning: Option "efield_outside_map" of section TPC is not supported and will be ignored.
  warnings.warn(f'Option "{key}" of section {s} is not supported'
```
Trying to look for a typo somewhere in how the `efield_outside_map` is defined.

**Review by ramirezdiego — APPROVED** [review]
I'm happy to merge now. Famous last words @HenningSE @WenzDaniel ?
